### PR TITLE
build: Add warning when missing postgres submodule during the build

### PR DESCRIPTION
**Pull request by fluz**
I forked the project and in my local repo, I wasn't able to compile the project and in my search, I found the solution in neon forum. After a PR discussion, I made a change in the makefile to alert the missing `git submodules update` step.

### CODE REVIEW

**Comment by shanyp** [comment]
@fluz thank you for your contribution

**Review by shanyp — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -120,11 +120,28 @@ make -j`sysctl -n hw.logicalcpu` -s
...
+env: /Users/userA/git/neon/vendor/postgres-v14/configure: No such file or directory
+make: *** [/Users/userA/git/neon/pg_install//build/v14/config.status] Error 127
+```
+Check if you have cloned the submodules with the patched PostgreSQL. To do this, check 
+if the folder `neon/vendor/postgres-v14` contains some files there. If not, execute the 
+following command:
+```bash
+# from https://community.neon.tech/t/problem-installing-neon-locally-on-fedora/258
+git submodule update --init --recursive --depth 2 --progress .
+```
```
Comment (koivunej) [inline README.md:144]: Above in the README there is already a suggestion of using `git clone --recursive` for this reason (lines 94, 111).

As an alternative idea: We could detect the absense of submodules in the makefile, and fix it for the user near: https://github.com/neondatabase/neon/blob/b1a11261527c740f8abb222bbcaf8852d62b9b17/Makefile#L73-L80

**Comment by shanyp** [comment]
@fluz are you planning to continue on this one?

**Comment by fluz** [comment]
Yes @shanyp, I plan to implement the suggestion from @koivunej and add a message in the makefile process. But I'm a little overloaded this week. But I will propose a new version of this PR in this weekend :-)

**Comment by shanyp** [comment]
Great ! thanks for contributing @fluz

File: README.md (hunk 1 above)
Comment (fluz) [reply README.md]: Hi @koivunej , your comment makes sense. I dropped my change in the Readme, and I added this warning in makefile.

**Comment by fluz** [comment]
@shanyp and @koivunej please take a look again in my PR 🙏🏾  :-)

File: README.md (hunk 1 above)
Comment (koivunej) [reply README.md]: This is looking great! I'll resolve this soon once I've finished a test run.

File: Makefile [hunk 2]
```diff
@@ -72,6 +72,10 @@ neon: postgres-headers walproposer-lib
 #
 $(POSTGRES_INSTALL_DIR)/build/%/config.status:
 	+@echo "Configuring Postgres $* build"
+	test -s $(ROOT_PROJECT_DIR)/vendor/postgres-$*/configure || { \
+		echo "\nPostgres submodule not found in $(ROOT_PROJECT_DIR)/vendor/postgres-$*/, execute "; \
+		echo "'git submodule update --init --recursive --depth 2 --progress .' in project root.\n"; \
+		exit 1; }
```
Comment (koivunej) [inline Makefile:78]: My makefile-fu is rusty, would be great to disable echoing.

Comment (koivunej) [reply Makefile:78]:
```suggestion
	@test -s $(ROOT_PROJECT_DIR)/vendor/postgres-$*/configure || { \
		echo "\nPostgres submodule not found in $(ROOT_PROJECT_DIR)/vendor/postgres-$*/, execute "; \
		echo "'git submodule update --init --recursive --depth 2 --progress .' in project root.\n"; \
		exit 1; }
```

**Comment by koivunej** [comment]
Thanks a lot for this @fluz!

**Pull request merged**